### DISCUSSION TITLE
Is it possible to open scene viewer without showing the 3d model with model-viewer?

### DISCUSSION THREAD

**Discussion by hd-esteban** [google/model-viewer#1860] [Q&A]
For some cases we would like to create a link to open directly scene viewer, so we would like to know if it is possible.

I noticed model-viewer component creates an anchor with a 'intent' command with some fallback parameters. 

is it possible to hide the model-viewer component but keep the link to open the AR functionality directly?

I tried with an anchor with the href attribute set with the intent command but it is not working, i also tried by handling the event and using another anchor to trigger the .click() method but it didn't work, does it have any restriction or something like that?

intent://subdomain.domain.com/file.glb?
link=https://domain/ar-example/ar-example.html
&title=3dmodel
#Intent;
scheme=http;
package=com.google.ar.core;
action=android.intent.action.VIEW;
S.browser_fallback_url=https://domain/ar-example.html#model-viewer-no-ar-fallback;
end

`<a href="intent://subdomain.domain.com/file.glb?link=https://domain/ar-example.html&title=3dmodel#Intent;scheme=http;package=com.google.ar.core;action=android.intent.action.VIEW;S.browser_fallback_url=https://domain/ar-example.html#model-viewer-no-ar-fallback;end"> AR </a>`

**Comment by cdata** [comment]
Hi @Esteban26 thanks for the questions.

🚨**Important disclaimer:** the implementation you are looking at in `<model-viewer>` is presently considered a private / undocumented Scene Viewer API and is subject to change at any time!

To answer your high-level question: it is possible to set `reveal="interaction"` on `<model-viewer>` and also use the `ar` attribute to show the AR button where available without loading the model right away. This will have the effect of displaying the poster and deferring the model until the user interacts with the poster. However, please also be aware that on Android devices without ARCore, `<model-viewer>` will still fallback to going fullscreen when the button is pressed.

To answer your low level question: the code you are looking at is the code that launches Scene Viewer. `<model-viewer>` doesn't have any special / restricted capabilities, but per the disclaimer above the API to launch Scene Viewer is considered private / undocumented and is subject to change without notice. `<model-viewer>` will be updated as quickly as possible when the API changes. **The easiest way to avoid a breaking change is to continue to launch Scene Viewer with `<model-viewer>` until the Scene Viewer API is publicly documented.**

In short: you probably just have a small typo or something in the syntax, keep trying!

**Comment by hd-esteban** [comment]
Thanks for your response @cdata :) I already found my issues

**Comment by hd-esteban** [comment]
intent://subdomain.domain.com/file.glb?
link=https://domain/ar-example/ar-example.html
&title=3dmodel
#Intent;
**scheme=https;**
package=com.google.ar.core;
action=android.intent.action.VIEW;
S.browser_fallback_url=https://domain/ar-example.html#model-viewer-no-ar-fallback;
**end;**

**Comment by hd-esteban** [comment]
@cdata is the **reveal** attribute a public parameter/attribute? I didn't see it at the documentation. Will this attribute be able in future versions?

https://googlewebcomponents.github.io/model-viewer/index.html

**Comment by cdata** [comment]
Sorry, your comment took me by surprise because I wrote the docs for that attribute ;)

It looks like the docs were accidentally reverted to an older version. I filed https://github.com/GoogleWebComponents/model-viewer/issues/549 to track this issue.

In the mean time, you can read the correct docs in the PR where `reveal` was added: #487

**Comment by cdata** [comment]
Here is a more direct link to the correct docs:

https://github.com/GoogleWebComponents/model-viewer/blob/076c380c85538b3b0b7a12f6965bc97a9a7035d7/index.html#L201-L208

**Comment by tehfonsi** [comment]
You can find the full documentation here: https://developers.google.com/ar/develop/java/scene-viewer#ar-only

**Comment by AytoMaximo** [comment]
Hi! I've tried to inspect this thread and #631 carefully, but still can't get the solution. 

Is there a straight answer can showing GLB model in AR be as simple as USDZ only by direct link? Or scanning a QR code for example? Just point your device's camera at QR code and be redirected directly to AR mode? 

I've tried the way with calling intent, but it also works only if user touch the link with his own finger, calling this `a href` with ajax or any other implementation of "autoclick" feature leads to fallback url. 

I remember the same issue with autoclicking default ar-button in model-viewer, console says such implementations are not permitted. God why?

**Comment by elalish** [comment]
You're running up against the browser security model. Entering AR either via WebXR or App intent is not allowed except by user activation. This means they have to actually click the AR button; you can't do it for them. You can use a custom AR button using our `slot` that covers the whole element to make it clear if you want.

**Comment by WadeMorrison** [comment]
Is it not possible to use the intent:// on IOS?

I basically have a product configurator made that you can configure on desktop, and then when you click a <button> "View In Room", a QR code will generate from the intent url so the configured product will launch automatically in AR instead of having to re-configure it all again on mobile and then finally click "View In Space".

However whenever I put the intent URL in, and scan the generated QR Code, my Iphone says "No Usable Data Found". So for now I just have it linked to the document.URL so it'll at least open the model-viewer on the client's phone to configure it.

Hopefully this makes sense? I've only got like a month's worth of history with HTML, CSS, JS so far so any help is greatly appreciated!

**Reply by elalish** [reply]
For all the privacy/security reasons stated above, you generally can't activate directly into AR. However, you should be able to still show the model pre-configured on mobile and let the user click view in your space. Depending on how much configuration you're doing, you may also try the piping-server approach that we use for this feature in our editor: https://modelviewer.dev/editor/
